From: Konstantin Khlebnikov <khlebnikov@openvz.org>
Date: Tue, 10 Mar 2009 14:31:18 +0300
Subject: [PATCH] pidns: zap ve process only when killing ve's init pid-ns

commit d876c93aa8cbd06f47e3669f5d8fcec1afc77ed4 upstream.

This prevents task genocide when zapping nested pid-ns in same ve,
and affects ve0 only.

Signed-off-by: Konstantin Khlebnikov <khlebnikov@openvz.org>
Signed-off-by: Pavel Emelyanov <xemul@openvz.org>
[bwh: Adjust context]
---
 kernel/pid_namespace.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/kernel/pid_namespace.c b/kernel/pid_namespace.c
index 4aee3b9..12f679e 100644
--- a/kernel/pid_namespace.c
+++ b/kernel/pid_namespace.c
@@ -339,7 +339,8 @@ void zap_pid_ns_processes(struct pid_namespace *pid_ns)
 
 
 #ifdef CONFIG_VE
-	zap_ve_processes(get_exec_env());
+	if (get_exec_env()->ve_ns->pid_ns == pid_ns)
+		zap_ve_processes(get_exec_env());
 #endif
 	return;
 }
